From 6530c66e28c930c51b5094a6040d690b3ae858e9 Mon Sep 17 00:00:00 2001
From: Michael Gratton <mike@vee.net>
Date: Sun, 25 Oct 2020 12:55:28 +1100
Subject: [PATCH 111/124] Geary.Imap.FolderSession: Fix null param critical
 with vala 0.50

Cancellable can be null, mark it as such - more fallout from
GNOME/vala#299

Fixes #1044
---
 src/engine/imap/api/imap-folder-session.vala | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/engine/imap/api/imap-folder-session.vala b/src/engine/imap/api/imap-folder-session.vala
index 8a2290cb..3b777ee6 100644
--- a/src/engine/imap/api/imap-folder-session.vala
+++ b/src/engine/imap/api/imap-folder-session.vala
@@ -85,8 +85,8 @@ private class Geary.Imap.FolderSession : Geary.Imap.SessionObject {
 
     public async FolderSession(ClientSession session,
                                Imap.Folder folder,
-                               Cancellable cancellable)
-        throws Error {
+                               GLib.Cancellable? cancellable)
+        throws GLib.Error {
         base(session);
         this.folder = folder;
         this.quirks = session.quirks;
-- 
2.29.2

